### PR TITLE
Refactor Move() Code

### PR DESCRIPTION
**Pull request by Arokha**
Cleans it up and makes it more well-documented. Several things were being done inconsistently between moving and forceMove, some things being called in some cases and not others.

Removes most overrides of Move(). If you need to know if you moved, use Moved(). If you just care about z-level changes for radios or something, use onTransitZ.

We've been playing with this on our server and haven't found issues except for one related to carrying light sources, for which the fix is included below.

### CODE REVIEW

**Comment by Arokha** [comment]
Fixed conflicts

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -1,403 +1,402 @@
...
+
+/mob/living/carbon/rejuvenate()
+	bloodstr.clear_reagents()
+	ingested.clear_reagents()
+	touching.clear_reagents()
+	..()
+
+/mob/living/carbon/Moved(atom/old_loc, direction, forced = FALSE)
+	. = ..()
+  if(src.nutrition && src.stat != 2)
```
Comment (Atermonera) [inline code/modules/mob/living/carbon/carbon.dm:38]: travis

Comment (Arokha) [reply code/modules/mob/living/carbon/carbon.dm]: GRUMBLE. Fixed.

File: code/modules/mob/living/silicon/robot/robot_movement.dm [hunk 2]
```diff
@@ -33,52 +33,52 @@
...
+						cleaned_human.wear_suit.clean_blood()
+						cleaned_human.update_inv_wear_suit(0)
+					else if(cleaned_human.w_uniform)
+						cleaned_human.w_uniform.clean_blood()
+						cleaned_human.update_inv_w_uniform(0)
+					if(cleaned_human.shoes)
+						cleaned_human.shoes.clean_blood()
+						cleaned_human.update_inv_shoes(0)
+					cleaned_human.clean_blood(1)
+					cleaned_human << "<font color='red'>[src] cleans your face!</font>"
```
Comment (Atermonera) [inline code/modules/mob/living/silicon/robot/robot_movement.dm:84]: to_chat()

**Pull request merged**